### PR TITLE
Implement Cart details sheet

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Implement Details Sheet for Cart and Checkout (part of Cart Inventory).

Add documents to Cart query to show in new Details Sheet.

Make changes to `FullscreenDialog` to be more flexible.

Add extra properties to UI Button to work as a link.

## Justify why they are needed

I'm using some placeholders for the data rows since we don't have it exposed in the API yet.

We should move out also the "confirm remove" dialog to a similar component but we can do that in the next step.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1259** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1259" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/1259?utm_source=stack-comment).

File: apps/store/src/components/CartInventory/CartEntryItem.tsx [hunk 1]
```diff
@@ -41,18 +42,25 @@ export const CartEntryItem = (props: Props) => {
             </Text>
           </div>
           <SpaceFlex space={0.25}>
-            <LinkButtonSecondary href="#">
-              {t('VIEW_ENTRY_DETAILS_BUTTON', { ns: 'cart' })}
-            </LinkButtonSecondary>
+            <DetailsSheetDialog {...cartEntry}>
+              <Dialog.Trigger asChild>
+                <Button variant="secondary" size="small">
+                  {t('VIEW_ENTRY_DETAILS_BUTTON', { ns: 'cart' })}
```
Comment (alebedev) [inline apps/store/src/components/CartInventory/CartEntryItem.tsx:48]: ns is redundant here, you've set it in useTranslation call above

File: apps/store/src/components/CartInventory/CartInventory.types.ts [hunk 2]
```diff
@@ -6,6 +7,7 @@ export type CartEntry = {
   cost: Money
   startDate?: Date
   pillow: { src: string; alt?: string }
+  documents: CartFragmentFragment['entries'][number]['variant']['documents']
```
Comment (alebedev) [inline apps/store/src/components/CartInventory/CartInventory.types.ts:10]: Would it be easier to use if we explicitly used ProductDocument fragment in GraphQL?

File: apps/store/src/components/CartInventory/DetailsSheetDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  const formatter = useFormatter()
+
+  return (
+    <FullscreenDialog.Root>
+      {children}
+
+      <FullscreenDialog.Modal
+        Footer={
+          <FullscreenDialog.Close asChild>
+            <Button type="button">Klar</Button>
```
Comment (alebedev) [inline apps/store/src/components/CartInventory/DetailsSheetDialog.tsx:29]: Localization?

**Review by alebedev — APPROVED** [review]
Great, feels closer to being launch ready!

**Pull request merged**